src/bacnet/basic/service/h_apdu.c

```diff
@@ -439,6 +439,14 @@ uint16_t apdu_decode_confirmed_service_request(uint8_t *apdu, /* APDU data */
         service_data->max_segs = decode_max_segs(apdu[1]);
         service_data->max_resp = decode_max_apdu(apdu[1]);
         service_data->invoke_id = apdu[2];
+        if (apdu[3] == 0x0c) /* Read property */
```

Comment:
please apply to ALL confirmed service requests, not just ReadProperty.

Reply:
Thanks. I hope I have fixed this correctly now.
The atmega168 port is fixed.